tokio/src/net/tcp/socket.rs

```diff
@@ -666,6 +666,7 @@ impl TcpSocket {
     /// socket must not have been connected prior to calling this function. This
     /// function is typically used together with crates such as [`socket2`] to
     /// configure socket options that are not available on `TcpSocket`.
+    /// It is left up to the user to set the socket in non-blocking mode.
```

Comment:
I wonder if this ought to be explicitly marked as a **Note** or **Warning** so that it's very obvious to users? Perhaps the warning also should note what will happen if users don't do this?

Reply:
Yeah, I agree.

Reply:
I applied your suggestion and created a more obvious section about this. I did the same with other docs containing the same warning and added missing examples.